src/middleware.rs

```diff
@@ -22,8 +22,8 @@ pub trait Middleware<State>: 'static + Send + Sync {
     ) -> BoxFuture<'a, crate::Result>;
 
     /// Set the middleware's name. By default it uses the type signature.
-    fn name(&self) -> &str {
-        std::any::type_name::<Self>()
+    fn name(&self) -> String {
```

Comment:
Calling attention to this signature change, which allows middlewares to return dynamic strings in their names, as used in https://github.com/http-rs/tide/pull/549/files#diff-c606b36cae12145a41d5fb07309603bfR108